tests/snappi_tests/pfc/test_valid_pfc_frame_with_snappi.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import pytest
+
+from tests.snappi_tests.pfc.files.helper import run_pfc_test
+from tests.common.helpers.assertions import pytest_require
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts,\
+    fanout_graph_facts # noqa F401
+from tests.common.snappi_tests.snappi_fixtures import snappi_api_serv_ip, snappi_api_serv_port,\
+    snappi_api, snappi_testbed_config # noqa F401
+from tests.common.snappi_tests.qos_fixtures import prio_dscp_map, all_prio_list, lossless_prio_list,\
+    lossy_prio_list # noqa F401
+from tests.common.snappi_tests.snappi_test_params import SnappiTestParams
+from tests.common.snappi_tests.common_helpers import packet_capture
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [pytest.mark.topology('tgen')]
+
+def test_valid_pfc_frame(snappi_api, # noqa F811
+                         snappi_testbed_config, # noqa F811
+                         conn_graph_facts, # noqa F811
+                         fanout_graph_facts, # noqa F811
+                         duthosts,
+                         rand_one_dut_hostname,
+                         rand_one_dut_portname_oper_up,
+                         lossless_prio_list, # noqa F811
+                         lossy_prio_list, # noqa F811
+                         prio_dscp_map): # noqa F811
+    """
+    Test if PFC Pause frame generated by device under test (DUT) is valid.
+
+    Topology:
+    tgen (1) -> DUT -> tgen (2)
+
+    Test steps:
+    1) Create congestion on ingress port of ixia (tgen 2).
```

Comment:
How can you create congestion with 1 sender and 1 receiver? I guess you create congestion by letting 1 send data traffic to 2, and 2 send PFC pause frames to DUT.

Reply:
Yes that's right